libs/sis-service/src/main/java/uk/gov/di/ipv/core/library/sis/service/SisService.java

```diff
@@ -1,29 +1,341 @@
+        this.auditService = AuditService.create(configService);
+        this.criResponseService = new CriResponseService(configService);
+        this.cimitUtilityService = new CimitUtilityService(configService);
+        this.evcsService = new EvcsService(configService);
+        this.votMatcher =
+                new VotMatcher(
+                        userIdentityService, new Gpg45ProfileEvaluator(), cimitUtilityService);
     }
 
     public SisGetStoredIdentityResult getStoredIdentity(
             ClientOAuthSessionItem clientOAuthSessionItem) {
+        // We want to get any identity in SIS so we ask for all vots.
         return sisClient.getStoredIdentity(
                 clientOAuthSessionItem.getEvcsAccessToken(),
-                clientOAuthSessionItem.getVtrAsVots(),
+                List.of(Vot.P0, Vot.P1, Vot.P2, Vot.P3),
                 clientOAuthSessionItem.getGovukSigninJourneyId());
     }
+
+    public void compareStoredIdentityWithStoredVcs(
+            ClientOAuthSessionItem clientOAuthSessionItem, AuditEventUser auditEventUser) {
+        SisGetStoredIdentityResult storedIdentityResult = null;
+        List<String> sisVcSignatures = new ArrayList<>();
+        List<String> evcsVcSignatures = new ArrayList<>();
+        Optional<VotMatchingResult.VotAndProfile> maximumMatchedVot = Optional.empty();
+
+        try {
+            String evcsAccessToken = clientOAuthSessionItem.getEvcsAccessToken();
+            String userId = clientOAuthSessionItem.getUserId();
+
+            // Get SIS details
+            storedIdentityResult = getStoredIdentity(clientOAuthSessionItem);
+
+            if (!storedIdentityResult.requestSucceeded()) {
+                throw new SisMatchException(
+                        FailureCode.SIS_ERROR,
+                        "Call to SIS service failed, no stored identity comparison can be made");
+            }
+
+            if (!storedIdentityResult.identityWasFound()) {
+                LOGGER.info(
+                        LogHelper.buildLogMessage(
+                                "No credential found in SIS so no comparison to do"));
+                return;
+            }
+
+            // Get stored signatures
+            sisVcSignatures = getSisSignatures(storedIdentityResult.identityDetails());
+
+            // Get EVCS details
+            var evcsCredentials = getEvcsVerifiableCredentials(userId, evcsAccessToken);
+
+            evcsVcSignatures =
+                    evcsCredentials.stream()
+                            .map(credential -> getVcSignature(credential.getVcString()))
+                            .toList();
+
+            maximumMatchedVot = calculateMaximumMatchedVot(evcsCredentials, userId);
+
+            // Compare VOTs
+            var evcsVot = maximumMatchedVot.isPresent() ? maximumMatchedVot.get().vot() : null;
+            if (storedIdentityResult.identityDetails().vot() != evcsVot) {
```

Comment:
Need to check we're comparing the right `vot` here - believe we should be comparing with the `vot` calculated by SIS which is a claim of the JWT itself (not the top-level `vot` returned in the response)